src/helpers/radix_sort.rs

```diff
@@ -0,0 +1,247 @@
+        .map(|x| helper.get_share_of_one() - *x)
+        .collect();
+    x_and_one_minus_x.append(&mut shares.to_vec());
+
+    // Step2: Calculate cumulative sum
+    let length = x_and_one_minus_x.len();
+    let mut cumulative_sum = Vec::with_capacity(length);
+    cumulative_sum.push(x_and_one_minus_x[0]);
+    for i in 1..length {
+        cumulative_sum.push(cumulative_sum[i - 1] + x_and_one_minus_x[i]);
+    }
+
+    // Step3: Consolidate step 1 and step 2 output
+    let mut mult_input = Vec::with_capacity(length);
+    for i in 0..cumulative_sum.len() {
+        mult_input.push((x_and_one_minus_x[i], cumulative_sum[i]));
+    }
+    mult_input
+}
+
+fn helper_gen_permutation_share(
+    mult_output: &[ReplicatedFp31SecretSharing],
+) -> Vec<ReplicatedFp31SecretSharing> {
+    let len = mult_output.len() / 2;
+
+    // Fold the result
+    let mut permutation_share = Vec::new();
+    (0..len).for_each(|j| {
+        permutation_share.push(mult_output[j] + mult_output[j + len]);
+    });
+    permutation_share
+}
+
+fn get_helper_specific_shares(
+    shares: &[ReplicatedShares],
+) -> Vec<Vec<ReplicatedFp31SecretSharing>> {
+    let mut helper_shares: Vec<Vec<ReplicatedFp31SecretSharing>> = Vec::new();
+    let mut helper1_share = Vec::new();
+    let mut helper2_share = Vec::new();
+    let mut helper3_share = Vec::new();
+
+    (0..shares.len()).for_each(|i| {
+        helper1_share.push(shares[i].0);
+        helper2_share.push(shares[i].1);
+        helper3_share.push(shares[i].2);
+    });
+
+    helper_shares.push(helper1_share);
+    helper_shares.push(helper2_share);
+    helper_shares.push(helper3_share);
+
+    helper_shares
+}
+
+#[cfg(test)]
+mod tests {
+
+    use crate::{
+        field::Fp31,
+        helpers::radix_sort::{helpers_do_multiplication, helpers_gen_permutations},
+        modulus_convert::{make_three, secret_share, ReplicatedFp31SecretSharing},
+    };
+
+    use super::{helpers_gen_mult_inputs, mpc_sort};
+
+    #[test]
+    fn replicated_share_sort() {
+        let input = [
+            (1, 0, 0),
+            (0, 0, 0),
+            (0, 1, 0),
+            (0, 0, 0),
+            (0, 0, 0),
+            (0, 0, 0),
+        ]
+        .to_vec();
+        let mut shares = Vec::new();
+        for iter in input {
+            shares.push(secret_share(iter.0, iter.1, iter.2));
+        }
+
+        let mpc_replicated_output = mpc_sort(&shares);
+        let expected_sort_output = [5_u128, 1, 6, 2, 3, 4].as_ref();
+
+        (0..mpc_replicated_output.len()).for_each(|i| {
+            assert_eq!(
+                mpc_replicated_output[i].0
+                    + mpc_replicated_output[i].1
+                    + mpc_replicated_output[i].2,
+                ReplicatedFp31SecretSharing::construct(
+                    Fp31::from(expected_sort_output[i]),
+                    Fp31::from(expected_sort_output[i])
+                )
+            );
+        });
+    }
+
+    #[test]
+    fn gen_mult_inputs() {
```

Comment:
I don't think we need to separately test this one. The test above is probably enough. I'm also not sure this particular division into sub-functions is the one we will stick with - so adding tests for each one will just make it harder to refactor.